modules/parquet/src/parquetjs/utils/decoders.ts

```diff
@@ -63,7 +63,7 @@ export async function decodeDataPages(
       continue;
     }
 
-    if (dictionary) {
+    if (dictionary.length) {
```

Comment:
Are we sure dictionary is defined? 
```suggestion
    if (dictionary?.length) {
```

Also it is still really confusing me that this file is in the `utils` directory. This file seems to contain the meat of the parquet decoding logic, not some minor utilities.

Reply:
1. Yes it is defined in 51 line: `let dictionary = options.dictionary || [];`
2. It was done in terms of refactoring parquet module. Need to think about moving it to another folder or do some renaming.